batcher/aligned-batcher/src/lib.rs

```diff
@@ -268,7 +268,48 @@ impl Batcher {
             });
         }
 
-        Ok(())
+        Err(BatcherError::EthereumSubscriptionError(
+            "No more blocks".to_string(),
+        ))
+    }
+
+    async fn get_next_block_with_retry(
+        &self,
+        stream: Arc<Mutex<SubscriptionStream<'_, Ws, Block<TxHash>>>>,
+        stream_fallback: Arc<Mutex<SubscriptionStream<'_, Ws, Block<TxHash>>>>,
+    ) -> Result<Block<H256>, BatcherError> {
+        retry_function(
+            || {
+                let stream = Arc::clone(&stream);
+                let stream_fallback = Arc::clone(&stream_fallback);
+                async move {
+                    let mut stream_lock = stream.lock().await;
+                    let mut stream_fallback_lock = stream_fallback.lock().await;
+                    self.get_next_block(&mut stream_lock, &mut stream_fallback_lock)
+                        .await
+                }
+            },
+            DEFAULT_MIN_DELAY,
+            DEFAULT_FACTOR,
+            DEFAULT_MAX_TIMES,
+        )
+        .await
+        .map_err(|_| BatcherError::EthereumSubscriptionError("No more blocks".to_string()))
+    }
+
+    async fn get_next_block(
+        &self,
+        stream: &mut SubscriptionStream<'_, Ws, Block<TxHash>>,
+        stream_fallback: &mut SubscriptionStream<'_, Ws, Block<TxHash>>,
+    ) -> Result<Block<H256>, RetryError<()>> {
+        tokio::select! {
+            Some(block) = stream.next() => Ok(block),
+            Some(block) = stream_fallback.next() => Ok(block),
+            else => {
+                warn!("Error fetching next block.");
+                Err(RetryError::Transient)
+            }
+        }
```

Comment:
the functionality to get new blocks works by a websocket subscription event, so we don't really need to make it retry. In the future it may be useful to add polling of new event blocks for better reliability than only the websocket events, and it will make sense to have a retry there, but for now let's just leave it like it is